lldb-eval/eval_test.cc

```diff
@@ -1061,3 +1116,71 @@ TEST_F(EvalTest, TestBitField) {
   EXPECT_THAT(Scope("abf").Eval("0 + b"), IsEqual("15"));
   EXPECT_THAT(Scope("abf").Eval("0 + c"), IsEqual("3"));
 }
+
+TEST_F(EvalTest, TestContextVariables) {
+  // Context variables don't exist yet.
+  EXPECT_THAT(EvalWithContext("$ctx", vars_),
+              IsError("use of undeclared identifier '$ctx'"));
+  EXPECT_THAT(Scope("s").EvalWithContext("$ctx", vars_),
+              IsError("use of undeclared identifier '$ctx'"));
+
+  EXPECT_TRUE(CreateContextVariable("$ctx", "13"));
+  EXPECT_THAT(EvalWithContext("$ctx", vars_), IsEqual("13"));
+  EXPECT_THAT(EvalWithContext("$ctx + 2", vars_), IsEqual("15"));
+  EXPECT_THAT(EvalWithContext("$ctx - s.a", vars_), IsEqual("3"));
+  EXPECT_THAT(EvalWithContext("ctx", vars_),
+              IsError("use of undeclared identifier 'ctx'"));
+  EXPECT_THAT(Scope("s").EvalWithContext("$ctx", vars_), IsEqual("13"));
+  EXPECT_THAT(Scope("s").EvalWithContext("$ctx + 2", vars_), IsEqual("15"));
+  EXPECT_THAT(Scope("s").EvalWithContext("$ctx - a", vars_), IsEqual("3"));
+  EXPECT_THAT(Scope("s").EvalWithContext("ctx", vars_),
+              IsError("use of undeclared identifier 'ctx'"));
+
+  // Context variable is a pointer.
+  EXPECT_TRUE(CreateContextVariable("$ptr", "s.ptr"));
+  // TODO(tsabolcec): Change the following line once the pointer
+  // comparison issue is fixed.
+  EXPECT_THAT(EvalWithContext("$ptr == s.ptr", vars_),
+              IsError("comparison of distinct pointer types"));
+  EXPECT_THAT(EvalWithContext("*$ptr", vars_), IsEqual("'h'"));
+  EXPECT_THAT(EvalWithContext("$ptr[1]", vars_), IsEqual("'e'"));
+  EXPECT_THAT(Scope("s").EvalWithContext("$ptr == ptr", vars_), IsError(""));
+  EXPECT_THAT(Scope("s").EvalWithContext("*$ptr", vars_), IsEqual("'h'"));
+  EXPECT_THAT(Scope("s").EvalWithContext("$ptr[1]", vars_), IsEqual("'e'"));
+
+  EXPECT_THAT(EvalWithContext("$ctx + *$ptr", vars_), IsEqual("117"));
+  EXPECT_THAT(Scope("s").EvalWithContext("$ctx + *$ptr", vars_),
+              IsEqual("117"));
+
+  // Context variables that were created in this test are visible from LLDB.
+  // Disable comparisons with LLDB to test subsets of created context variables.
+  this->compare_with_lldb_ = false;
```

Comment:
Can you put this and everything below in a separate `TEST_F` function? As you've noticed `compare_with_lldb_` changes the behavior for the current test, so ideally we'd like to minimize the scope.

Reply:
Done.